### DISCUSSION TITLE
about AutoResponse

### DISCUSSION THREAD

**Discussion by DMMSHA256** [ORelio/Minecraft-Console-Client#2432] [Q&A]
matches.ini
[Match]
match=abcde
action=send 1
actionprivate=send /tell $u Hello!
actionother=log detected "hi" message
ownersonly=false
cooldown=0
“abcde”Can be detected,and actionother Can be executed.But action can't work
Please help me, thank you..

**Comment by MousePak** [comment]
Check > https://github.com/MCCTeam/Minecraft-Console-Client/issues/2433
This might be a bug 

Must concur, I was unable to get this simple check to work 

```
<> [%s] [Num] [Rank] MousePak: abcde
[MCC] [AutoRespond] Running action: log detected "hi" message
[MCC] [AutoRespond] detected "hi" message
```

Further testing also commented out **actionother** mcc returned nothing too
[Match]
match=abcde
action=send 11
actionprivate=send /tell $u Hello!
#actionother=log detected "hi" message
ownersonly=false
cooldown=0

Sending in channel /// sending to player
```
<> [%s] [1.03k] [Legend] MousePak: abcde
<> [skyblock] [me -> TestAcct] abcde
```
No /tell command or send 1 triggered

**Comment by MousePak** [comment]
After further investigation the error you have will most likely be your server chat being non default

<> [%s] [Num] [Rank] MousePak: abcde
[MCC] [AutoRespond] Running action: send 1
<> [%s] [Num] [Rank] TestAcct:  1


Im no regex pro so you will need to do a bit of research yourself 

Using the server chat format im using the following needed to be changed in config
Each server can have very unique chat styles so a cut-n-paste will likely fail you
```
[ChatFormat]
Builtins = false                            # MCC support for common message formats. Set "false" to avoid conflicts w$
UserDefined = true                          # Whether to use the custom regular expressions below for detection.
Public = "^.+] ([a-zA-Z0-9_]+): (.+)$"
Private = "^.*[(.*) ->.+] (.*)$"
```

**Reply by DMMSHA256** [reply]
Thank you. I have solved this problem with regex and regular expression, but my regular expression cannot be used correctly in [ChatFormat], and can only be detected correctly in [matches. ini]

**Reply by MousePak** [reply]
The chat format used on my server had a lot of [ ] groups so my regex was very generic for public / private captures. Just tried searching for player names and all text after : must be chat.
<> [%s] [809] [Legend**] Player1 [tag]**: Hm
<> [%s] [809] [Legend] Player2 [tag]: **Actually a decent idea**
<> [%s] [1.01k] [II] [Epic] Player3: stupid and crazy

Once I got the correct regex for ChatFormat the match.ini you shared worked. Oddly other match's I had which used regex also started working that I ignored in the past.

My regex is pure guess work but try tweaking in regex101 until you get something which game chat.. after that matching should start working. Good luck.

**Comment by Sasubaysa** [comment]
@MousePak  how can I reach you ? , İ need help

**Comment by MousePak** [comment]
@Sasubaysa seen you have folked the official repository, i'll be unlikely to assist. If related to mcc may I suggest logging an issue or discussion here as I dont share contact details directly.

**Reply by Sasubaysa** [reply]
https://github.com/MCCTeam/Minecraft-Console-Client/discussions/1903#discussioncomment-5460676 




Would you please look at the comments of this post, I have stated my problem.
